app/controllers/Checkout.scala

```diff
@@ -45,6 +45,7 @@ object Checkout extends Controller {
           yield BadRequest(views.html.checkout.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
       },
       userData => {
+        CheckoutService.processSubscription(userData, AuthenticationService.authenticatedUserFor(request))
```

Comment:
Ok, I am fine in doing this as a temporary measure, but I am starting to think that `CheckoutService.processSubscription`, as well as most of the other calls we do to external applications should not return values of type `Future[Option[T]]`, but just of `Future[T]` (or maybe Either[Throwable, T]).